x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/all_rules/batch_actions.tsx

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiContextMenuItem } from '@elastic/eui';
+import React from 'react';
+import * as i18n from '../translations';
+import { TableData } from '../types';
+import { Action } from './reducer';
+import { deleteRulesAction, enableRulesAction, exportRulesAction } from './actions';
+
+export const getBatchItems = (
+  selectedState: TableData[],
+  dispatch: React.Dispatch<Action>,
+  closePopover: () => void,
+  kbnVersion: string
+) => {
+  const containsEnabled = selectedState.some(v => v.activate);
+  const containsDisabled = selectedState.some(v => !v.activate);
+  const containsLoading = selectedState.some(v => v.isLoading);
+
+  return [
+    <EuiContextMenuItem
+      key={i18n.BATCH_ACTION_ACTIVATE_SELECTED}
+      icon="checkInCircleFilled"
+      disabled={containsLoading || !containsDisabled}
+      onClick={async () => {
+        closePopover();
+        const deactivatedIds = selectedState.filter(s => !s.activate).map(s => s.id);
+        await enableRulesAction(deactivatedIds, true, dispatch, kbnVersion);
```

Comment:
nit: Above you use object structuring: `enableRules({ ids, enabled, kbnVersion });` and here you use normal parameters. Just pointing it out. Was expecting both to be of the same type.

No changes needed.

Reply:
Yeah, the API calls were structured to accept objects/named params, whereas both `actions` and `batch_actions` simply accept normal params. There's consistency between each type at least, but when I re-visit this I can move actions/batch_actions over to named params as well.